test/integration/provider/azure.go

```diff
@@ -0,0 +1,182 @@
+package provider
+
+/**
+	Orphaned Resources
+	- VMs:
+		Describe instances with specified tag name:<cluster-name>
+		Report/Print out instances found
+		Describe volumes attached to the instance (using instance id)
+		Report/Print out volumes found
+		Delete attached volumes found
+		Terminate instances found
+	- Disks:
+		Describe volumes with tag status:available
+		Report/Print out volumes found
+		Delete identified volumes
+**/
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+
+	provider "github.com/gardener/machine-controller-manager-provider-azure/pkg/azure"
+	"github.com/gardener/machine-controller-manager-provider-azure/pkg/spi"
+	v1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	"github.com/gardener/machine-controller-manager/pkg/util/provider/driver"
+)
+
+// AdditionalResourcesCheck check for orphan network interfaces and triggers their deletion
+func additionalResourcesCheck(
+	clients spi.AzureDriverClientsInterface,
+	resourceGroup,
+	tagName,
+	tagValue string,
+) error {
+	ctx := context.TODO()
+
+	networkInterfaces, err := clients.GetNic().List(ctx, resourceGroup)
+	if err != nil {
+		return err
+	}
+	for _, networkInterface := range networkInterfaces.Values() {
+		if value, ok := networkInterface.Tags[tagName]; ok && tagValue == *value {
+			fmt.Println(networkInterface.Name)
```

Comment:
we can add a println over here like "Orphan network interfaces are:\n" and then print the name of nics